### ISSUE TITLE
Simplify API by exporting factory

### ISSUE THREAD

**Issue by davidkpiano** [sass-eyeglass/eyeglass#46]
This is a bit of a major change, but I believe that incorporating Eyeglass modules into a project will have less friction by simply exporting a factory function instead of a raw `Eyeglass` object.

**OLD API:**
```js
var sass = require('node-sass');
var Eyeglass = require('eyeglass');
var sassOptions = { ... };
var eyeglass = new Eyeglass(sassOptions); // boilerplate

eyeglass.assets("assets/images", "images", "public/images"); // imperative

// eyeglass.sassOptions() is also boilerplate
sass.render(eyeglass.sassOptions(), function(err, result) { ... });
```

**NEW (proposed) API:**
```js
var sass = require('node-sass');
var eyeglass = require('eyeglass');
var sassOptions = { ... };
var assets = [
  {source: 'assets/images', http: 'images', build: 'public/images'} 
]; // declarative and clear

sass.render(eyeglass(sassOptions, { assets: assets }), function(err, result) { ... });
```

So the arguments would simply be `eyeglass(sassOptions, eyeglassOptions)` and would return a POJO equivalent to `Eyeglass.prototype.options`. Both bits of boilerplate are eliminated, and it also allows the developer to do this shorthand:

```js
var sassOptions = { ... };
var eyeglassOptions = require('eyeglass')(sassOptions);
```

What do you think? I believe it's more intuitive and idiomatic.

**Comment by Jakobo** [comment]
@chriseppstein @eoneill @joannajw

**Comment by scottdavis** [comment]
I dig it.

**Comment by eoneill** [comment]
I like the proposal.

**Comment by davidkpiano** [comment]
@chriseppstein any thoughts on this? I know the `assets` part might be different now but I do believe the API can be simplified.

**Comment by chriseppstein** [comment]
The API did change but I am open to having a way to pass asset declarations in as options because I think this will help grunt integration.

The current API is:

```js
var eyeglass = new Eyeglass({});
eyeglass.assets.addSource("assets", {pattern: "images/**/*"});
```

The `addSource` method passes through to the [AssetPathEntry constructor](https://github.com/sass-eyeglass/eyeglass/blob/master/lib/assets.js#L17-L31).

So it seems like at the highest level, you're asking for a way to specify this via options instead of calling methods. :shrug: ok. We can just iterate over some option and call the method for you.

```js
var sass = require('node-sass');
var eyeglass = require('eyeglass').decorate;
var sassOptions = {
  assetSources: [
    {directory: "assets", pattern: "images/**/*"}
  ]
}

sass.render(eyeglass(sassOptions), function(err, result) { ... });
```

The second aspect of this request is to separate node-sass options from eyeglass options and handle them separately. To my eye this makes the code less user friendly. But I'm happy to discuss this on a separate issue.

**Comment by chriseppstein** [comment]
@davidkpiano does this seem ok?

**Comment by eoneill** [comment]
For those still interested in this, please review #94

**Comment by chriseppstein** [comment]
This is fixed in 0.8.
